### DISCUSSION TITLE
How to Configure Magnetic Reed Switch with Tasmota & NodeMCU

### DISCUSSION THREAD

**Discussion by Gosokan** [arendst/Tasmota#17390] [Q&A]
Hello, I have Magnetic Switch Sensor and NodeMCU v3. 
When the magnetic field approaches the sensor, the led on the sensor lights up, 
How to configure Tasmota to display ON or OFF status based on these conditions?

Magnetic reed switch is connected to D7 (GPIO 13) in NodeMCU.

![tasmota-magnetic-reed-switch-on_off](https://user-images.githubusercontent.com/13587422/207510830-6965b494-a524-4c9d-8624-4831e1a0c77c.png)

**Comment by barbudor** [comment]
As of today there is no easy way to display the status of an input switch in tasmota

What you see as on/off are status of the relay output

To get an input you need to define the GPIO as a Switch. If you also have a relay on your system and don't want the switch input to command the relay, you need to decouple the switch from the relay (see chapter Buttons and Switch).
Assuming it is Switch 1, you can achieve that with 'switchmode 1 15'
The you will get messages from tasmota each time the input change state. These can be processed in rules for action, but not display in web GUI easily

To customize the web GUI you need to invest time in learning Scripting-Language (or on ESP32 the Berry language)

**Reply by barbudor** [reply]
A basic way to report the switch input would be to define a unused gpio as a dummy relay so when the switch input change state, it change the relay status on the web
In that case don't use switchmode described above as you need the switch and the relay to remain coupled

There is no way to prevent someone click on the web button which will de sync the display from the state of the input until next state change

**Reply by Gosokan** [reply]
Finally the reed switch can function as I want after I reset the tasmota configuration. 
Then, I reconfigured again by selecting the module; 
Configure Module => Sonoff Basic (1)
GPIO 14 => Switch 2
Then I connected pin D0 on the magnetic reed switch module to D5 (GPIO14) on NodeMCU.

The result:
When I bring the magnet closer to the reed switch module, the status of Tasmota's WebUI changes from OFF to ON, and when i move the magnet away, the status goes back to OFF.

Regarding to prevent someone click on toggle button on the webui that's fine. Because I always put tasmota webui behind external reverse proxy. So webui only show ON/OFF status and information button. Any url query is blocked, except /?m=1

Anyway thank for help.

**Comment by JyunWei-Su** [comment]
You can use a MCP230xx (MCP23008/MCP23017) module if you have lot of reed switchs.
I wirte come simple guide about it. [MCP230XX-Tasmota](https://github.com/JyunWei-Su/MCP230XX-Tasmota)

**Reply by sfromis** [reply]
Two things....

When you want to show your project, instead of posting in an old thread which was already answered, we also have the "Show and tell" category here, designed for such posts. Your are welcome to post there.

Also, as a starting point I'd recommend using the greatly enhanced Mode 2 for MCP230xx, where you can configure the pins to simply act as standard Tasmota Relay/Button/Switch/Output pins, without needing any special handling after that. This is easier to work with than the older approach with the `Sensor29` command and special rules.
